daemon/update_linux.go

```diff
@@ -23,7 +23,7 @@ func toContainerdResources(resources container.Resources) libcontainerd.Resource
 	r.CpusetCpus = resources.CpusetCpus
 	r.CpusetMems = resources.CpusetMems
 	r.MemoryLimit = uint64(resources.Memory)
-	if resources.MemorySwap > 0 {
+	if resources.MemorySwap != 0 {
 		r.MemorySwap = uint64(resources.MemorySwap)
```

Comment:
Wondering what's the right thing to do here for `-1`. IIRC, the kernel itself accepts `-1` as a special value (even though it otherwise takes an `uint`.